lib/resolve-config.js

```diff
@@ -44,18 +45,31 @@ module.exports = (
     assets: assets ? castArray(assets) : assets,
     milestones: milestones ? castArray(milestones) : milestones,
     successComment,
-    proxy: getProxyConfiguration(defaultedGitlabUrl, HTTP_PROXY, HTTPS_PROXY),
+    proxy: getProxyConfiguration(defaultedGitlabUrl, HTTP_PROXY, HTTPS_PROXY, NO_PROXY),
   };
 };
-
-function getProxyConfiguration(gitlabUrl, HTTP_PROXY, HTTPS_PROXY) {
+ 
+function bypassProxy(NO_PROXY, gitlabUrl) {
+  if (NO_PROXY === '*') {
+    return true;
+  }
+  const hostName = gitlabUrl.split('://')[1].split(':')[0];
+  const noProxyEntries = NO_PROXY.split(',');
+  return noProxyEntries.some((proxyConfig) => {
+    return hostName.endsWith(proxyConfig.toLowerCase().trim().replace(/^\./, ""));
+  });
+}
```

Comment:
I think it's a little bit more complex than this. Maybe have a look at the implementation in the pretty common https://www.npmjs.com/package/proxy-from-env package.

Reply:
Reworked on using proxy-from-env package